src/llmcompressor/transformers/compression/helpers.py

```diff
@@ -240,3 +249,104 @@ def calculate_offload_device_map(
+        exhaustive_targets=exhaustive_targets,
+        exhaustive_ignore=exhaustive_ignore,
+    )
+
+
+def is_sparse_compression_target(
+    module: torch.nn.Module, sparsity_threshold: float, sparsity_structure: str
+) -> bool:
+    """
+    :param module: module to check
+    :param sparsity_threshold: threshold for sparsity
+    :param sparsity_structure: sparsity structure to check against
+    :return: whether or not the module is a target for sparsity compression,
+        i.e True if it is sparse and follows the sparsity structure, else False
+    """
+    return (
+        hasattr(module, "weight")
+        and tensor_sparsity(module.weight) >= sparsity_threshold
+        and tensor_follows_mask_structure(tensor=module.weight, mask=sparsity_structure)
+    )
+
+
+def _get_sparse_targets_ignore_dicts(
+    module: torch.nn.Module, sparsity_structure: str, sparsity_threshold: float
+) -> Tuple[Dict[str, List[str]], Dict[str, List[str]]]:
+    """
+    Get sparse targets and ignore dictionaries
+
+    :param module: module to check
+    :param sparsity_structure: sparsity structure to check against
+    :param sparsity_threshold: threshold for sparsity
+    :return: tuple of exhaustive targets and ignore dictionaries
+    """
+    exhaustive_targets = defaultdict(list)
+    exhaustive_ignore = defaultdict(list)
+
+    for name, submodule in iter_named_leaf_modules(module):
+        submodule_type = module_type(submodule)
+        is_target = is_sparse_compression_target(
+            module=submodule,
+            sparsity_threshold=sparsity_threshold,
+            sparsity_structure=sparsity_structure,
+        )
+        target_dict = exhaustive_targets if is_target else exhaustive_ignore
+        target_dict[submodule_type].append(name)
+    return exhaustive_targets, exhaustive_ignore
+
+
+def _reduce_targets_and_ignores_into_lists(
+    exhaustive_targets: Dict[str, List[str]], exhaustive_ignore: Dict[str, List[str]]
+) -> Tuple[List[str], List[str]]:
+    """
+    Reduces the targets and ignores dictionaries into lists
+
+    :param exhaustive_targets: dictionary of target layers, must contain all
+        targetted layers in the model
+    :param exhaustive_ignore: dictionary of ignore layers, must contain all
+        ignored layers in the model
+    :return: tuple of reduced target and ignore layers
+    """
+
+    targets, ignore = [], []
+    all_submodule_types = set(exhaustive_targets.keys()).union(
+        set(exhaustive_ignore.keys())
+    )
+    for submodule_type in all_submodule_types:
+        curr_targets = exhaustive_targets.get(submodule_type, [])
+        curr_ignores = exhaustive_ignore.get(submodule_type, [])
+
+        if len(curr_targets) > len(curr_ignores):
+            targets.append(submodule_type)
+            ignore.extend(curr_ignores)
+        else:
+            ignore.append(submodule_type)
+            targets.extend(curr_targets)
+
+    return targets, ignore
```

Comment:
Rather than the ignore list containing all non sparse modules, I would vote we only put something in the ignore list if it conflicts with something in the targets list. For instance, if we  are targeting all Linear layers but a few, the exceptions would go in the ignore list. But if a whole type is ignored (ie layer norms) they would be left out of the ignore list. This would then match how we treat ignore in the quantization modifier

Reply:
+1 to the above. Also how we're integrating/parsing on the vLLM side.

Reply:
That is a very good suggestion, @Satrat, I'm adding that in now!